src/argon2.lua

```diff
@@ -63,6 +61,9 @@ local _M     = {
     _URL     = "https://github.com/thibaultCha/lua-argon2-ffi",
 }
 
+local encoded_len = function(salt_len, hash_len)
+    return 40 + (math.ceil(salt_len / 3) + math.ceil(hash_len / 3)) * 4
```

Comment:
argon2 calculates the encoded length in [this way](https://github.com/P-H-C/phc-winner-argon2/blob/master/src/run.c#L126), and [argon2_encodedlen](https://github.com/P-H-C/phc-winner-argon2/blob/master/src/argon2.c#L431). I agree `40` is not good, it should be calculated in argon2's way.

I think hash length should be an option, like iterations and parallelism. Actually it's an option in argon2 [command-line utility](https://github.com/P-H-C/phc-winner-argon2#command-line-utility).

If you want the original constant `ENCODED_LEN` and `HASH_LEN` and to let the user override it, please document it in the usage. It took me some time to figure out the "Encoding failed" problem.